Add --update-config flag to deploy function command

RafalWilinski opened the pull request:
## What did you implement:

Closes https://github.com/serverless/serverless/issues/4042

As issue suggests, this PR uses only [UpdateFunctionConfiguration](http://docs.aws.amazon.com/lambda/latest/dg/API_UpdateFunctionConfiguration.html#SSS-UpdateFunctionConfiguration-request-Timeout), so it does zero changes to CloudFormation template.

## How can we verify it:

Deploy simple service:
```yml
service: aws-nodejs-tests

provider:
  name: aws
  runtime: nodejs6.10
  stage: dev

functions:
  hello:
    handler: handler.hello
    name: ${self:provider.stage}-lambdaName
```

And then change some config values & run `serverless deploy function -f hello -u` 
```yml
service: aws-nodejs-tests

provider:
  name: aws
  runtime: nodejs6.10
  stage: dev

functions:
  hello:
    handler: handler.hello
    name: ${self:provider.stage}-lambdaName
    description: Updated desc
    memorySize: 256
    timeout: 6
    environment:
      VARIABLE: varrrr
      VARTWO: two
```

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

eahefnawy commented:
Works like a charm!! Thanks tons @RafalWilinski ! your contributions are priceless 😊 

I'm thinking maybe update config should happen by default on `sls deploy function`, what you think? cc @pmuens

pmuens commented on lib/plugins/aws/deployFunction/index.js:
```diff
@@ -5,6 +5,7 @@ const _ = require('lodash');
 const crypto = require('crypto');
 const path = require('path');
 const fs = require('fs');
+const omitEmpty = require('omit-empty');
```
Can't we use https://lodash.com/docs/4.17.4#omit for that?

pmuens commented on package.json:
```diff
@@ -113,6 +113,7 @@
     "minimist": "^1.2.0",
     "moment": "^2.13.0",
     "node-fetch": "^1.6.0",
+    "omit-empty": "^0.4.1",
```
Can't we use `_.omit` from `lodash` for that? This we we'd save on external dependency.

RafalWilinski replied:
Unfortunately, from my experience `_.omit` does not work for deep values.

pmuens commented on lib/plugins/aws/deployFunction/index.js:
```diff
@@ -66,6 +72,38 @@ class AwsDeployFunction {
   }
 
+  updateFunctionConfiguration() {
+    const params = omitEmpty({
```
Does this include config which is defined on a the `provider` config as well?

See: https://serverless.com/framework/docs/providers/aws/guide/serverless.yml/

RafalWilinski replied:
Good catch!

pmuens commented on docs/providers/aws/guide/deploying.md:
````diff
@@ -78,7 +78,7 @@ This deployment method does not touch your AWS CloudFormation Stack.  Instead, i
 ```
 
-**Note:** You can always enforce a deployment using the `--force` option.
+**Note:** You can always enforce a deployment using the `--force` option or `--update-config` to change function configuration.
````
I'd add a separate `**Note:**` section below that since the `--force` and `--update-config` flags provide two distinct functionalities.

RafalWilinski replied:
Sure

eahefnawy commented:
Alright! thanks for the feedback @pmuens 😊 ... Let's do that then @RafalWilinski 👍

RafalWilinski commented:
@pmuens  
- removed `omit-empty` dependency
- changed `updateFunctionConfiguration` behavior to take `service` and `provider` into account
- changed `--update-config` flag behavior. By default, `serverless deploy function` will update Lambda config while `--update-config` will just update config without deploying new code.
